### PR TITLE
Improved Documentation on how to write a changelog

### LINKED ISSUE #3854: Improve documention on how to write a changelog
Currently, the dev guide has only a brief explanation on how to write a changelog entry. We should consider including the more detailed instructions [in this file](https://github.com/sunpy/sunpy/blob/master/changelog/README.rst) in the dev guide or simply linking to that file.

### PR DESCRIPTION
**Pull request by aribalam**
### Description

Added a more comprehensive instruction on how to write a changelog. Taken reference from the file `changelog/README.rst`

Fixes #3854

### CODE REVIEW

**Comment by wtbarnes** [comment]
Should we be copying and pasting this or can we somehow link to the file that's already in the root of the repo? It'd be better to avoid having to maintain the same information in two places as this increases the chances of something going stale.

**Comment by aribalam** [comment]
@wtbarnes I will revert all the changes and add a link to the file.

File: docs/dev_guide/maintainer_workflow.rst [hunk 1]
```diff
@@ -187,11 +187,15 @@ Updating and Maintaining the Changelog
 
 The changelog will be read by users, so this description should be aimed at SunPy users instead of describing internal changes which are only relevant to the developers.
 
+Instructions on how to write a changelog is given `here`_
```
Comment (wtbarnes) [inline docs/dev_guide/maintainer_workflow.rst]:
```suggestion
Instructions on how to write a changelog can be found `here`_
```

**Comment by wtbarnes** [comment]
Ironically, needs a changelog 😁

File: docs/dev_guide/maintainer_workflow.rst [hunk 2]
```diff
@@ -187,7 +187,7 @@ Updating and Maintaining the Changelog
 
 The changelog will be read by users, so this description should be aimed at SunPy users instead of describing internal changes which are only relevant to the developers.
 
-Instructions on how to write a changelog is given `here`_
+Instructions on how to write a changelog can be found `here`_
```
Comment (dpshelio) [inline docs/dev_guide/maintainer_workflow.rst]: Now that we are with this... we shouldn't use `here` on any of our links!
Read why ['click here' links are not good for accessibility on this Berkeley guide](https://webaccess.berkeley.edu/ask-pecan/click-here).

File: docs/dev_guide/maintainer_workflow.rst [hunk 3]
```diff
@@ -188,10 +188,16 @@ Updating and Maintaining the Changelog
...
+You do not need to update this file as we use `towncrier`_ to update our changelog.
+This is built and embedded into our documentation.
+
+Towncrier will automatically reflow your text, so it will work best if you stick to a single paragraph, but multiple sentences and links are OK and encouraged.
+You can install towncrier and then run `towncrier --draft` if you want to get a preview of how your change will look in the final release notes.
+
+Instructions on how to write a changelog can be found `here`_.
 
 .. _towncrier: https://pypi.org/project/towncrier/
+.. _here: https://github.com/sunpy/sunpy/blob/master/changelog/README.rst
```
Comment (nabobalis) [inline docs/dev_guide/maintainer_workflow.rst]:
```suggestion
```

File: docs/dev_guide/maintainer_workflow.rst [hunk 4]
```diff
@@ -188,10 +188,16 @@ Updating and Maintaining the Changelog
...
 The current changelog is kept in the file "CHANGELOG.rst" at the root of the repository.
-We use `towncrier`_ to update our changelog.
-This is built and embedded into our documentation and instructions are in "sunpy/changelog/README.rst".
+You do not need to update this file as we use `towncrier`_ to update our changelog.
+This is built and embedded into our documentation.
+
+Towncrier will automatically reflow your text, so it will work best if you stick to a single paragraph, but multiple sentences and links are OK and encouraged.
+You can install towncrier and then run `towncrier --draft` if you want to get a preview of how your change will look in the final release notes.
+
+Instructions on how to write a changelog can be found `here`_.
```
Comment (nabobalis) [inline docs/dev_guide/maintainer_workflow.rst]:
```suggestion
`Instructions on how to write a changelog. <https://github.com/sunpy/sunpy/blob/master/changelog/README.rst>`__.
```

File: docs/dev_guide/maintainer_workflow.rst (hunk 2 above)
Comment (nabobalis) [reply docs/dev_guide/maintainer_workflow.rst]: How about now?

**Pull request merged**